blobfuse2-gen1-artifacts.yaml

```diff
@@ -0,0 +1,189 @@
+# Pipeline to build artifact for gen1 compliant adlsgen1fuse
+
+# Do not trigger this pipeline automatically
+trigger: none
+pr: none
+
+stages:
+
+  # ---------------------- Build adlsgen1fuse here and copy binary to artifact ------------------------------
+  - stage: BuildAdlsGen1Fuse
+    jobs:
+      - job: AdlsGen1FuseBuild
+
+        # Define on what all OS we want to run this Pipeline
+        strategy:
+          matrix:
+            Ubuntu-18:
+              container_image: 'ubuntu-18.04'
+              build_type: Release
+        pool:
+          vmImage: $(container_image)
+
+        variables:
+        - group: NightlyBlobFuse
+        - name: ROOT_DIR
+          value: '$(System.DefaultWorkingDirectory)'
+        - name: WORK_DIR
+          value: '$(System.DefaultWorkingDirectory)/BlobFuse-ADLSGen1'
+
+        steps:
+          - checkout: none
+
+          # Install all dependencies
+          - script: |
+              sudo apt-get update --fix-missing
+              sudo apt-get install pkg-config cmake libcurl4-gnutls-dev libgnutls28-dev uuid-dev libgcrypt20-dev libboost-all-dev gcc g++ -y
+              sudo apt-get install libfuse-dev -y
+            displayName: "Basic Tools Setup"
+
+          - script: |
+              sudo apt update
+              sudo apt upgrade
+              sudo apt-get install pkg-config clang-9 -y
+              curl https://sh.rustup.rs -sSf | sh -s -- -y
+              rustup install nightly
+              rustup install nightly-2021-09-01
+              rustup default nightly-2021-09-01-x86_64-unknown-linux-gnu
+              sudo apt install llvm
+              llvm-config --prefix
+            displayName: "Rust Setup"
+            
+          - script: |
+              git clone https://$(DEVOPS_UNAME):$(DEVOPS_PWD)@dev.azure.com/azstorage/_git/BlobFuse-ADLSGen1
+            displayName: "Checkout code"
+            workingDirectory: $(ROOT_DIR)
+
+          # List commits consumed for this build
+          - script: |
+              echo "Below commits were done in last 12 hours : " `date` 
+              git --no-pager log --since="12 hours ago" --stat
+            displayName: "List Commits"
+            workingDirectory: $(WORK_DIR)
+            
+          # Build adlsgen1fuse binary
+          - script: |
```

Comment:
why are we building the binary ourselves? can't we just take the published binary?

Reply:
gen1fuse standard binary comes packaged inside the ML workspace and that does not run independently. To solve this gen1 mount issue Nara created another workspace where we took all the required code from ML workspace and created a rust based binary out of it. So there is no standard release of datasetfuse as such. We have our own repo which we build and create the gen1fuse binary and later package it with the blobfuse binary.